api/envoy/extensions/filters/http/composite/v3/composite.proto

```diff
@@ -53,8 +53,9 @@ message ExecuteFilterAction {
   config.core.v3.TypedExtensionConfig typed_config = 1
       [(udpa.annotations.field_migrate).oneof_promotion = "config_type"];
 
-  // Dynamic configuration of filter obtained via extension configuration discovery
-  // service.
+  // Dynamic configuration of filter obtained via extension configuration discovery service.
+  // When composite filter is in upstream filter chain, the ``dynamic_config``
+  // can not be set. TBD: Refactor cluster manager init sequence to Support this.
```

Comment:
To support the  dynamic_config when composite filter is in upstream filter chain, the cluster manager init sequence need to be slightly refactored to make it work.  This is due to dynamic upstream filter creation needs the cluster manager instance to be created first.  We will have a follow up PR to address this.

Reply:
Ack, thanks, can you add this as TODO(#Pr number) or the like, so it's clear this is being followed up on and where to track progress? Plan sounds good, as we're only making things less restrictive and hence don't break anything.

Reply:
Or file an OSS issue and todo on it?

Reply:
I created an OSS issue for this: https://github.com/envoyproxy/envoy/issues/33218

In this issue, I proposed a solution to make it work. The solution is just a few lines change in the cluster_manager_ initializing sequence, and this solution can make the composite filter with dynamic action_config work in upstream filter chain based on the test.  I will send out a PR for that change shortly. We can discuss that solution in that PR.

Reply:
This is the draft PR to refactor cluster manager init sequence:
https://github.com/envoyproxy/envoy/pull/33221
If folks agree the general idea of the approach, I will work on to get it into shape.

After #33221, and #33013 committed, then we just need to have some minor change in composite/action.cc to support dynamic action_config in composite filter.   We can discuss that then. But at this moment, I think we should get the static composite filter working in upstream filter chain first.

Reply:
Sure, just reference this in the comment and remove when thee PRs land.

Reply:
Done